worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Worker.scala

```diff
@@ -389,10 +390,14 @@ private[celeborn] class Worker(
   private def cleanup(expiredShuffleKeys: JHashSet[String]): Unit = synchronized {
     expiredShuffleKeys.asScala.foreach { shuffleKey =>
       partitionLocationInfo.getAllMasterLocations(shuffleKey).asScala.foreach { partition =>
-        partition.asInstanceOf[WorkingPartition].getFileWriter.destroy()
+        val fileWriter = partition.asInstanceOf[WorkingPartition].getFileWriter
+        fileWriter.destroy(new IOException(
+          s"Destroy FileWriter ${fileWriter} caused by shuffle ${shuffleKey} expired."))
       }
       partitionLocationInfo.getAllSlaveLocations(shuffleKey).asScala.foreach { partition =>
-        partition.asInstanceOf[WorkingPartition].getFileWriter.destroy()
+        val fileWriter = partition.asInstanceOf[WorkingPartition].getFileWriter
+        fileWriter.destroy(new IOException(
+          s"Destroy FileWriter ${fileWriter} caused by shuffle ${shuffleKey} expired."))
```

Comment:
@waitinfuture Should we add  ` FileWriter ${fileWriter}` here

Reply:
I think we'd better do so to be more specific.